version_template.py

```diff
@@ -107,14 +102,29 @@ def buildNumber():
 
 
 def currentCommit():
+    import os
     from subprocess import check_output
-    return check_output('git log --pretty=oneline -n 1 -- $(pwd)', shell=True).split()[0]
+
+    return check_output(['git', 'log', '--pretty=oneline', '-n', '1', os.getcwd()]).split()[0]
 
 
 def dirty():
+    """
+    Returns True if the code has uncommitted diffs.
+
+    :return: bool
+    """
     from subprocess import call
-    return 0 != call('(git diff --exit-code '
-                     '&& git diff --cached --exit-code) > /dev/null', shell=True)
+    import os
+
+    with open(os.devnull, 'w')  as FNULL:
+        # return code is 1 if there are still changes relative to the index
+        # otherwise return code is 0
+        retcode1 = call(['git', 'diff', '--exit-code'], stdout=FNULL)
+    # return code is 1 if there are still changes staged for the next commit relative to HEAD
+    # otherwise return code is 0
+    retcode2 = call(['git', 'diff', '--cached', '--exit-code'])
```

Comment:
This is not equivalent to the shell's `&&`, which does boolean short-circuit.

Reply:
I was thinking it would produce the same effect (excepting that I missed that it could negative values), but I've nested the two now so that it should be equivalent.